sandy/core/cov.py

```diff
@@ -1649,6 +1886,20 @@ def get_eig(cov, sort=True):
     return E, V
 
 
+def get_eig_sparse(cov, sort=True):
+    try:
+        E, V = sps.linalg.eig(cov)
+    except:
+        E, V = scipy.linalg.eig(cov.toarray())

```

Comment:
Why try/except? Should we add it also to get_eig?

Reply:
Because depending on the matrix cov, sometimes scipy.sparse is not optimize for that matrix and raise an error saying to move to scipy.linalg.eig.

Reply:
What exception do you get? You could work around a specific exception,  say `TypeError`, using `except TypeError`.

Reply:
To avoid rewriting parts of code, can we use:
```
except Exception:
    return self.eig()
```

Reply:
For the moment i will leave this part as it is, because sps.linalg.eig doesnt exits, so it was always doing scipy.linalg.eig(cov.toarray())